MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1401,32 +1403,57 @@ void MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::
 #else
       bankDir = (bankDir).expand(selectedfPath.parent().toString());
 #endif
-
       if (!splitBaseName.empty() && splitBaseName.size() > 3) {
         std::string foc_file = splitBaseName[0] + "_" + splitBaseName[1] + "_" +
                                splitBaseName[2] + "_" + splitBaseName[3];
         std::string strBankDir = bankDir.toString();
-        updateFittingDirVec(strBankDir, foc_file);
+        updateFittingDirVec(strBankDir, foc_file, false);
+
+        // add bank to the combo-box and list view
+        addBankItems(splitBaseName, focusedFile);
+        runNoVec.clear();
+        runNoVec.push_back(splitBaseName[1]);
+        if (!m_fittingMutliRunMode)
+          addRunNoItem(runNoVec, false);
       }
+      // assuming that no directory is found so look for number
       // if run number length greater
     } else if (focusedFile.count() > 4) {
-      // if given a run number instead
-      updateFittingDirVec(m_focusDir, strFocusedFile);
-    } else {
+      if (strFocusedFile.find("-") != std::string::npos) {
```

Comment:
More than one range is probably not needed, but we need to make sure that if confused users enter unexpected strings it will behave well, either give an error or use only the first range found.